docs/manual/file_mode.md

```diff
@@ -0,0 +1,95 @@
+
+## Format
+
+The format of an RPMIO mode string is as follows:
+
+```
+<mode>[flags].<type>
+```
+
+The `mode` parameter is mandatory and contains a sequence of letters as
+described in `fopen(3)`.  For payload configuration, only `w` is allowed.
+
+The `type` parameter is also mandatory and specifies the compression type
+(*compressor*) to be used.  Currently, the following compressors are supported:
+
+| Type		| Description					    |
+|---------------|---------------------------------------------------|
+|`gzdio`	| gzip ([flags](#gzip-flags))			    |
+|`bzdio`	| bzip2 ([flags](#bzip2-flags))			    |
+|`xzdio`	| lzma ([flags](#xz-lzma-flags))		    |
+|`lzdio`	| lzma alone (legacy) ([flags](#lzma-alone-flags))  |
+|`zstdio`	| zstd ([flags](#zstd-flags))			    |
+|`ufdio`	| uncompressed					    |
+
+The `flags` parameter is optional and contains compressor-specific options,
+most commonly a compression *level* or the *number of threads*.  For the actual
+supported values, consult the given compressor's section below.  For some
+real-world examples, see [Example Modes](#example-modes).
+
+#### GZIP Flags
+
+These can include a compression level (from 0 to 9) and a strategy (e.g. `h`
+for Huffman-only compression).  See the description of `gzopen()` in
+`/usr/include/zlib.h` for more details.
+
+#### BZIP2 Flags
+
+These can include a compression level (from 1 to 9) and `s` (enables "small
+mode").
+
+#### XZ (LZMA) Flags
+
+These can include a compression level (from 0 to 9) and `T` followed by the
+number of threads to be used during compression, where `T0` (or just `T`) means
+that the optimal number of threads should automatically be chosen based on the
+`%{getncpus}` macro.
+
+#### LZMA (Alone) Flags
+
+These can include a compression level (from 0 to 9).
+
+#### ZSTD Flags
+
+These can include a compression level (see `-#` in `zstd(1)` for the supported
+values), `T` followed by the number of threads to be used during compression,
+where `T0` means that the optimal number of threads should automatically be
+chosen based on the `%{getncpus}` macro, and `L` followed by a window size that
+enables long distance matching (see `--long` in `zstd(1)` for the supported
+values).
+
+### Example Modes
+
+These are some example mode strings suitable for the payload:
+
+| Mode          | Description                               |
+|---------------|-------------------------------------------|
+|`w9.gzdio`	| gzip level 9, RPM payload's default	    |
+|`w9.bzdio`	| bzip2 level 9, bzip2's default	    |
+|`w6.xzdio`	| xz level 6, xz's default		    |
+|`w7T16.xzdio`	| xz level 7 using 16 threads		    |
+|`w7T0.xzdio`	| xz level 7 using `%{getncpus}` threads    |
+|`w7T.xzdio`	| xz level 7 using `%{getncpus}` threads    |
+|`w6.lzdio`	| lzma-alone level 6, lzma's default	    |
+|`w3.zstdio`	| zstd level 3, zstd's default		    |
+|`w19T8.zstdio`	| zstd level 19 using 8 threads		    |
+|`w7T0.zstdio`	| zstd level 7 using `%{getncpus}` threads  |
```

Comment:
Replace %{getncpus} references with broad stroke "auto detected", see above.

I'd add a variant without the explicit compression level, eg 'w.xzdio', because that's legit too (IIRC) and just means "use whatever the default".

Reply:
Makes sense, thanks!

Reply:
And yes, that use is valid, too, I even tried it out with gdb. Basically only the `w` letter is required, a dot and a compression type.